api/container/containerv2/ingress.go

```diff
@@ -41,23 +71,69 @@ type SecretDeleteConfig struct {
 
 // SecretUpdateConfig secret update request
 type SecretUpdateConfig struct {
-	Cluster   string `json:"cluster" description:"name of secret" binding:"required"`
-	Name      string `json:"name" description:"name of secret" binding:"required"`
-	Namespace string `json:"namespace" description:"namespace of secret" binding:"required"`
-	CRN       string `json:"crn" description:"crn of the certificate in certificate manager"`
+	Cluster        string        `json:"cluster" description:"name of secret" binding:"required"`
+	Name           string        `json:"name" description:"name of secret" binding:"required"`
+	Namespace      string        `json:"namespace" description:"namespace of secret" binding:"required"`
+	CRN            string        `json:"crn" description:"crn of the certificate in secret manager"`
+	FieldsToAdd    []FieldAdd    `json:"add" description:"fields to add to secret"`
+	FieldsToRemove []FieldRemove `json:"remove" description:"fields to remove from secret"`
+}
+
+// Instance struct holding details for a single instance
+type Instance struct {
+	Cluster         string `json:"cluster" description:"id of cluster"`
+	Name            string `json:"name" description:"name of instance"`
+	CRN             string `json:"crn" description:"crn of the instance"`
+	SecretGroupID   string `json:"secretGroupID" description:"ID of the secret group where secrets will be stored"`
+	SecretGroupName string `json:"secretGroupName" description:"name of the secret group where secrets will be stored"`
+	CallbackChannel string `json:"callbackChannel" description:"callback channel of the instance"`
+	UserManaged     bool   `json:"userManaged" description:"true or false. Used to show which certs and secrets are system generated and which are not"`
+	IsDefault       bool   `json:"isDefault" description:"true or false. Used to show which instance subdomains certificates are uploaded into"`
+	Type            string `json:"type" description:"designates instance type as either certificate manager instance or secrets manager instance"`
+	Status          string `json:"status" description:"Used to show the status indicating if the instance is registered to the cluster or not"`
+}
+
+// Instances struct for a secret array
+type Instances []Instance
+
+// InstanceRegisterConfig the instance register request
+type InstanceRegisterConfig struct {
+	Cluster       string `json:"cluster" description:"id of cluster" binding:"required"`
+	CRN           string `json:"crn" description:"crn of the instance"`
+	IsDefault     bool   `json:"isDefault" description:"true or false. Used to show which instance subdomains certificates are uploaded into"`
+	SecretGroupID string `json:"secretGroupID" description:"ID of the secret group where secrets will be stored"`
+}
+
+// InstanceDeleteConfig the instance delete request
```

Comment:
do we want this to be delete or InstanceUnregisterConfig to be consistent with the register terminology?

Reply:
I kept it consistent with our internal model. I can change it if we think unregister is better